### PR TITLE
self-hosted core in overview table

### PR DESCRIPTION
**Pull request by MichaelTrestman**
This was a direct product request:
https://cockroachlabs.slack.com/archives/C01PHNMUFLN/p1692291727879799?thread_ts=1692281978.236079&cid=C01PHNMUFLN

https://cockroachlabs.atlassian.net/browse/DOC-8593

### CODE REVIEW

**Comment by kathancox** [comment]
As discussed in the slack thread, it might be good to prefix Enterprise with "Self-Hosted" here to differentiate. Otherwise LGTM.

File: src/current/v23.1/security-reference/security-overview.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ docs_area: reference.security
 
 Learn more: [Integrate {{ site.data.products.dedicated }} with Satori]({% link {{ page.version.version }}/satori-integration.md %})
 
-### Self-Hosted
+### Self-Hosted Core
```
Comment (abhinavg6) [inline src/current/v23.1/security-reference/security-overview.md:21]: I think this should remain as-is, as it's about the overall SH offering.

**Review by abhinavg6 — CHANGES_REQUESTED** [review]
Requesting one change, thanks.

**Review by abhinavg6 — APPROVED** [review]
LGTM

**Pull request merged**